OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -40,7 +41,8 @@ public void Killed(Actor self, AttackInfo e)
 			if (self.World.SharedRandom.Next(100) > info.Chance)
 				return;
 
-			if (info.DeathType != null && e.Warhead != null && !info.DeathType.Intersect(e.Warhead.DamageTypes).Any())
+			var warhead = e.Warhead as DamageWarhead;
+			if (info.DeathType != null && e.Warhead != null && !info.DeathType.Intersect(warhead.DamageTypes).Any())
```

Comment:
This (and the other code like it) need to be checking `warhead != null` to avoid a NRE if `e.Warhead` isn't a DamageWarhead.